packages/flutter/lib/src/material/text_field.dart

```diff
@@ -835,6 +851,11 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
   FocusNode? _focusNode;
   FocusNode get _effectiveFocusNode => widget.focusNode ?? (_focusNode ??= FocusNode());
 
+  MaxLengthEnforcement get _effectiveMaxLengthEnforcement => LengthLimitingTextInputFormatter.defaultMaxLengthEnforcement(
+    widget.maxLengthEnforcement,
+    platform: Theme.of(context).platform,
```

Comment:
Considering someone set the platform to the theme, and IMO it should work on TextField because they both in the material package.

Reply:
According to the documentation, `ThemeData.platform` should override the platform mechanics (haptic feedback, etc.), from that point of view reading from `ThemeData.platform` makes sense. But in this particular case it feels a bit weird. If you set `ThemeData.platform` to iOS on Android, it's going to infer a `MaxLengthEnforment` value that doesn't work well on Gboard.

Reply:
I'm okay with both solutions, because it sounds like a normal case when you manually switch it. So if we use default target platform for all cases, both arguments the methods have could be removed. I remember that we don't recommend static getter in the code?

Reply:
Nvm using `Theme.of(context).platform` seems to make more sense, now that I understand we also take the platform's `maxLength` behavior into account.

Reply:
I thought there are nothing left because these reviews are resolved 🤣 I'll bring up a new PR for improvement.